server/game/core/ability/PlayCardFromHandAction.ts

```diff
@@ -0,0 +1,22 @@
+import * as CostLibrary from '../../costs/CostLibrary';
+import { IActionTargetResolver } from '../../TargetInterfaces';
+import { Card } from '../card/Card';
+import { PlayType } from '../Constants';
+import { ICost } from '../cost/ICost';
+import { PlayCardAction } from './PlayCardAction';
+
+export class PlayCardFromHandAction extends PlayCardAction {
```

Comment:
@AMMayberry1 I'm drawing a blank on what we could do here besides renaming this. I assume we would need a new play type for playing from the Discard (TIE Silencer) and for Deck (such as Ezra Bridger). Would there be some sort of way to make this like templated like PlayCardAction<Location> (I don't know Typescript well enough to have the syntaxes down or know what it can do)

Reply:
Yeah sorry, thinking on it more I think the changes will be small enough that we don't need to account for them. I was thinking we could just use this class with some additional input parameters to control where it's legal to be played from but on reflection that might not end up being the best solution so we can just ignore that for now.